joomla/joomla-cms
# [4.0] Flags are country flags, not language flags

When installing a language, the corresponding content language is automatically created and also an image ("flag") associated with it.

However the logic for the language flags applied there is no proper logic at all: The issue is that there exists no language flags, they are country (or district) flags.

When a new official language is created by our translation teams, we manually have to upload a new image matching the new language code. For example for the german dialects this was done in a PR (see #11868).

Original PR was #12014. Feel free to read through the lengthy discussion there.

### Summary of Changes
This PR suggest to change the logic and assign the flags based on the country code in our language codes. Eg for de-CH (Swiss German) it would take the image ch.gif instead of de-ch.gif.
To still allow language specific images, it will first look for an existing file with the language code and fall back to the country code.

This PR also removes all now unneeded (since duplicated) files and renames some:

33 files are renamed to their corresponding country code.
75 files are deleted because they were duplicates and are no longer needed.
6 files are renamed to a name which already existed with a wrong image before. Those are si.gif, ca.gif, br.gif, be.gif, at.gif and af.gif. This accounts for a small B/C break for people who used those images before, as they will now change. Solution is simple: The user needs to select the image fresh in the content language.

### Testing Instructions

* Install a language and check that the created content languages have proper flags for the language / country they are referring.

### Documentation Changes Required
None that I am aware of.
